wagtail/wagtailsearch/backends/base.py

```diff
@@ -270,14 +281,19 @@ def search(self, query_string, model_or_queryset, fields=None, filters=None,
             for prefetch in prefetch_related:
                 queryset = queryset.prefetch_related(prefetch)
 
-        # Check operator
-        if operator is not None:
-            operator = operator.lower()
-            if operator not in ['or', 'and']:
-                raise ValueError("operator must be either 'or' or 'and'")
-
-        # Search
-        search_query = self.query_class(
-            queryset, query_string, fields=fields, operator=operator, order_by_relevance=order_by_relevance
-        )
-        return self.results_class(self, search_query)
+        self.check_operator(operator)
+
+        return (queryset, query_string), {
+            'fields': fields, 'operator': operator,
+            'order_by_relevance': order_by_relevance}
+
+    def search(self, query_string, model_or_queryset, fields=None, filters=None,
```

Comment:
Well… To me it makes things clearer and more extendable.
Instead of adding yet another argument, couldn’t it be the occasion to make it an `Autocomplete` class according to your RFC?

I guess we need to do `PlainText` and `MatchAll` before, then I’ll modify this pull request accordingly.